internal/cluster/cluster.go

```diff
@@ -1399,8 +1399,12 @@ func (m *Manager) setConnectivityMajorityGroupsForClusterInternal(cluster *commo
 		majorityGroups[cidr] = majorityGroup
 	}
 
+	primaryMachineCIDR := ""
+	if network.IsMachineCidrAvailable(cluster) {
+		primaryMachineCIDR = network.GetMachineCidrById(cluster, 0)
+	}
```

Comment:
Is there any reason to limit to a single MachineNetwork CIDR here?
This already will not work for dual stack clusters, because CreateL3MajorityGroup() is called for both IPv4 and IPv6, but this only passes the IPv4 MachineNetwork.
#6071 will allow multiple machine networks of each address family as well.

I would prefer to pass all of the MachineNetworks and check if each IP is in any of them.

Reply:
Gotcha, I modified it to check across multiple machine networks.